### PR TITLE
add a minimal and naive way to show balances across active chains

### PR DESCRIPTION
**Pull request by prasincs**
I wanted to be able to show the various balances across the different chains. I'm only checking what's available and not things like what osmosis LPs have or staked balances. I also added some validation around adding chains that was a TODO item.

Usage:

```
build/lens crosschain bank balances <default chain address>
```

It also takes a `--combined` flag to show you a "global" albeit flawed view of your balances that strips the name of IBC-channels, etc.

and it will print the balances for various tokens in the chains that you have added with `lens chains add <name>` Once that's done, it will sum up the responses by the denom. In case of IBC tokens, I'm ignoring the channels and just taking the last part of the identifier and used that as enum. For example, `transfer/channel-229/ubtsg` just becomes `ubtsg``
One benefit of this is to be able to see current token balances regardless of IBC status to be checked.

### CODE REVIEW

**Comment by Zygimantass** [comment]
Not sure if taking the last part of the identifier is a very good idea, since tokens with different channels are non-fungible.

**Comment by marbar3778** [comment]
on board with the cross chain command. is there a way to wrap the normal bank query in a loop of all the chains in the registry? this would avoid duplicated code.

File: cmd/chains.go [hunk 1]
```diff
@@ -69,17 +70,36 @@ func fetchRegistryChains(ctx context.Context) ([]string, error) {
...
+
 func cmdChainsAdd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:     "add [[chain-name]]",
 		Args:    cobra.MinimumNArgs(1),
 		Aliases: []string{"a"},
 		Short:   "add configraion for a chain or a number of chains from the chain registry",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			// TODO: input validation, maybe fetch the registry chains and validate the chain name
+			ctx := context.Background()
```
Comment (jackzampolin) [inline cmd/chains.go:89]:
```suggestion
			ctx := cmd.Context()
```

File: cmd/crosschainbank.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cobra"
+)
+
+// crosschainBankCmd represents the command to get balances across chains
+func crosschainBankCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "crosschainbank",
```
Comment (jackzampolin) [inline cmd/crosschainbank.go:15]: Lets make this a top level `crosschain` (`Aliases: []string{"cc", "kriskross", "cchain", "coolchain"}`) with subcommands for each of the modules below? Love the idea but think we need better command structure here.

File: cmd/crosschainbank.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+				keyNameOrAddress = ""
+				address          sdk.AccAddress
+				err              error
+			)
+			if len(args) == 0 {
+				keyNameOrAddress = cl.Config.Key
+			} else {
+				keyNameOrAddress = args[0]
+			}
+			if cl.KeyExists(keyNameOrAddress) {
```
Comment (jackzampolin) [inline cmd/crosschainbank.go:52]: Theres a helper function for this now. I can make this change.

File: cmd/crosschainbank.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	}
+
+	cmd.AddCommand(
+		getEnabledChainbalancesCmd(),
+	)
+
+	return cmd
+}
+
+func getEnabledChainbalancesCmd() *cobra.Command {
```
Comment (jackzampolin) [inline cmd/crosschainbank.go:27]: Reposting from TG: I don't think that having the default be just a list of balances is ideal. Would be nice to break the results out by chain (i.e. `map[chain-name][]string` with the strings being rendered coins from the account) and have the feature for a top level balance view (i.e. `map[denom]sdk.Int` or `[]string` rendered) behind a feature flag.

File: main.go [hunk 5]
```diff
@@ -15,8 +15,13 @@ limitations under the License.
...
 
-import "github.com/strangelove-ventures/lens/cmd"
+import (
+	"log"
+
+	"github.com/strangelove-ventures/lens/cmd"
+)
 
 func main() {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
```
Comment (jackzampolin) [inline main.go:25]: oh what is this!?

Comment (prasincs) [reply main.go:25]: oops, forgot to delete it, I wanted to see filenames in logs/errors. it turned out to not be useful. I'll remove it

**Comment by prasincs** [comment]
> is there a way to wrap the normal bank query in a loop of all the chains in the registry? this would avoid duplicated code.

yeah, but I looked at the duplication and it's pretty small.. we can definitely review it once the remaining functionality is completed?

File: main.go (hunk 5 above)
Comment (jackzampolin) [reply main.go]: Oh cool, just never seen that annotation before!

**Pull request merged**